sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/service/write/db/DeleteDocumentQueryBuilderTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.write.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+import io.stargate.bridge.grpc.Values;
+import io.stargate.bridge.proto.QueryOuterClass.Query;
+import io.stargate.sgv2.docsapi.DocsApiTestSchemaProvider;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import java.util.Map;
+import javax.inject.Inject;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@QuarkusTest
+@TestProfile(DeleteDocumentQueryBuilderTest.Config.class)
+class DeleteDocumentQueryBuilderTest {
+
+  private static final int MAX_DEPTH = 4;
+  private static final DocsApiTestSchemaProvider SCHEMA_PROVIDER =
+      new DocsApiTestSchemaProvider(MAX_DEPTH);
+  private static final String KEYSPACE_NAME = SCHEMA_PROVIDER.getKeyspace().getName();
+  private static final String COLLECTION_NAME = SCHEMA_PROVIDER.getTable().getName();
```

Comment:
I think eventually the cleaner solution would be to inject `DocumentProperties` into `DocsApiTestSchemaProvider`. I can make the change everywhere if we agree.

Reply:
If you have an idea it would be great to optimize.. In addition, maybe this should be connected with a fixed profile that defines the `maxDepth`. Like having a single profile for smaller `maxDepth`, be 4 or whatever.. That would give us:

* less code duplication for test setup
* faster tests, as for each different test profile Quarkus have to restart the context?

Reply:
Implemented in "Refactor unit tests" commit.